src/gcovr/formats/gcov/parser/json.py

```diff
@@ -197,9 +197,9 @@ def _parse_file_node(
             )
         for index, condition in enumerate(line.get("conditions", [])):
             linecov.insert_condition_coverage(
-                index,
                 ConditionCoverage(
                     str(data_fname),
+                    conditionno=index,
```

Comment:
How does this new parameter affect merging of conditions from the previous problem that was encountered?

Reply:
For now it's only moved from the function `insert_condition_coverage` to the `ConditionCoverage` class that all classes track their id. This is a compatible change but for your problem I want to create a test in a follow up PR. If I write to if conditions in one line I should be able to force the error you received because the current implementation wants to merge them.